tsdb/index.go

```diff
@@ -557,9 +557,10 @@ func IntersectSeriesIDIterators(itr0, itr1 SeriesIDIterator) SeriesIDIterator {
 
 	// Create series id set, if available.
 	if a := NewSeriesIDSetIterators([]SeriesIDIterator{itr0, itr1}); a != nil {
+		ss := a[0].SeriesIDSet().And(a[1].SeriesIDSet())
```

Comment:
Does `a[0] == itr0` and `a[1] == itr1` here? If so, could we add a comment saying so? I think it'll make it more obvious why the `Close()` calls have to happen after this

Reply:
i added comments. please check if it matches your suggestion.